### PR TITLE
FINERACT-1742: Introduce Lombok to dataqueries module

### PR DESCRIPTION
**Pull request by hmg65**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java [hunk 1]
```diff
@@ -135,10 +144,6 @@ private void adjustColumnTypes() {
         }
     }
 
-    public boolean isNamed(final String columnName) {
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java:138]: Why delete this one?

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java [hunk 2]
```diff
@@ -283,18 +264,6 @@ public boolean isCodeLookupDisplayType() {
         return "CODELOOKUP".equalsIgnoreCase(this.columnDisplayType);
     }
 
-    public boolean isMandatory() {
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java:286]: Why delete this one?

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java [hunk 3]
```diff
@@ -283,18 +264,6 @@ public boolean isCodeLookupDisplayType() {
...
-        return !isOptional();
-    }
-
-    public boolean isOptional() {
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java]: Why delete this one?

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java [hunk 4]
```diff
@@ -283,18 +264,6 @@ public boolean isCodeLookupDisplayType() {
...
-        return this.isColumnNullable;
-    }
-
-    public boolean hasColumnValues() {
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java]: Why delete this one?

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java [hunk 5]
```diff
@@ -305,14 +274,6 @@ public boolean isColumnValueAllowed(final String match) {
         return allowed;
     }
 
-    public boolean isColumnValueNotAllowed(final String match) {
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java]: Why delete this one?

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java [hunk 6]
```diff
@@ -305,14 +274,6 @@ public boolean isColumnValueAllowed(final String match) {
...
-        return !isColumnValueAllowed(match);
-    }
-
-    public boolean isColumnCodeNotAllowed(final Integer match) {
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java]: Why delete this one?

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java [hunk 7]
```diff
@@ -1203,11 +1203,11 @@ private CommandProcessingResult updateDatatableEntry(final String dataTableName,
 
         final GenericResultsetData grs = retrieveDataTableGenericResultSetForUpdate(appTable, dataTableName, appTableId, datatableId);
 
-        if (grs.hasNoEntries()) {
+        if (grs.getData().isEmpty()) {
             throw new DatatableNotFoundException(dataTableName, appTableId);
         }
 
-        if (grs.hasMoreThanOneEntry()) {
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java:1210]: Why deleted the original method?

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java (hunk 7 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java:1210]: i have directly implemented it

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java (hunk 6 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java]: fixed

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java (hunk 5 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java]: fixed

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java (hunk 4 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java]: fixed

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java (hunk 3 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java]: fixed

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java (hunk 2 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java:286]: fixed

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java (hunk 1 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/data/ResultsetColumnHeaderData.java:138]: directly implemented it at the usage

**Pull request closed without merging**